### PR TITLE
DDF-1463 Standard UI Map Layer Preferences.

### PR DESCRIPTION
**Pull request by shane-voisard**
@stustison @pklinef @jlcsmith @michaelmenousek @wmcnalli 

- works for both openlayers and cesium; saved changes
  reflected in big map(s).

- modal dialog presents mini-map to change layer:
    - z-order, via drag/drop
    - alpha, via slider
    - show/hide, via checkbox

- dialog workflow allows:
    - "bulk" edits
    - close w/o save
    - restore defaults.

- includes change for base modal destroy logic.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/382" target="_blank">ReviewNinja

**Comment by stustison** [comment]
Can you link to screenshots as well?

File: catalog/ui/search-ui/standard/src/main/webapp/js/controllers/cesium.controller.js [hunk 1]
```diff
@@ -20,27 +21,29 @@ define(['underscore',
...
-            CT: Cesium.CesiumTerrainProvider,
-            AGS: Cesium.ArcGisImageServerTerrainProvider,
-            VRW: Cesium.VRTheWorldTerrainProvider,
-            SI: Cesium.SingleTileImageryProvider
-        };
+        var imageryProviderTypes = LayerCollectionController.imageryProviderTypes;
+
+        var CesiumLayerCollectionController = LayerCollectionController.extend({
+            initialize: function () {
+                this.listenTo(wreqr.vent, 'layer:pref:reorder:bigMap', this.reIndexAll);
```
Comment (stustison) [inline catalog/ui/search-ui/standard/src/main/webapp/js/controllers/cesium.controller.js]: To make it easier to track events, we try to have the first part of the event be the module that the event is thrown from. (This is going to change in later versions of the UI, but that is how it is for now.)

**Comment by shane-voisard** [comment]
default-layers
visibility-changed-no-save
z-order-changed-no-save

**Comment by michaelmenousek** [comment]
1. Change "Z-Order" to "Order"
2. Change "Save" button to be blue, and have it appear first, followed by reset, and then cancel.  No need for Close
3. Make the bar of the slider lighter, so I can see it.
4.  Make the text vertically centered in the row.  The arrows icon is making the row too tall.  Looks like you need to shrink it.

**Comment by shane-voisard** [comment]
@michaelmenousek I have made all the L&F changes you requested, which I am prepared to push, but I think we need to clarify the workflow of the dialog. Currently:

- each tab's buttons are independent, so that saving and restoring only affect the tab shown.
- the save button turns from blue to red to indicate that a change will be lost unless it is clicked.
- restore is like any other edit, which must be saved before it takes affect.
- neither save nor restore dismiss the modal.

Is this how you want it to work?

**Comment by michaelmenousek** [comment]
I hope to avoid unnecessary code by using balsamiq for UI mockups before things get coded up.

I think the close button can be moved to an X in the corner of the modal.  You can change the name of "Save" to "Apply" which probably makes more sense given the behavior you're describing.  Also, if you want it to change colors it should change from grey to blue.  Red is for errors.

Summarizing,
* each tab's buttons are independent, so that saving and restoring only affect the tab shown.
 - yeah that's fine.  I think you mean reset not restore though.
* the save button turns from blue to red to indicate that a change will be lost unless it is clicked.
 - okay, but make it go grey to blue.
* restore is like any other edit, which must be saved before it takes affect.
 - yes but again this is reset to default, not "restore" right?
* neither save nor restore dismiss the modal.
 - that's okay with me since there are two tabs to deal with.

File: catalog/ui/search-ui/standard/src/main/webapp/js/controllers/cesium.controller.js (hunk 1 above)
Comment (shane-voisard) [inline catalog/ui/search-ui/standard/src/main/webapp/js/controllers/cesium.controller.js]: Fixed.

**Comment by shane-voisard** [comment]
color-prefs
layer-prefs

Screen shots above reflect changes requested by @michaelmenousek 

- 'X' button in top right corner of modal.
- 'Close' button removed.
- 'Save' button renamed 'Apply'
- 'Apply' button cycles from grey -> blue -> grey to represent editing workflow.
- order of buttons from left to right is now 'Apply', 'Reset Defaults', 'Cancel'
- layer picker styling:
    - 'Z-Order' column renamed 'Order'
    - column headers centered
    - sliders more visible
    - reduced row height

- Also changed user model to ensure consistent, predictable data flow.

**Comment by michaelmenousek** [comment]
LGTM thanks @shane-voisard

**Comment by shane-voisard** [comment]
Pushed small fix to support destroy of spectrum color bindings. Includes small merge conflict resolution.

**Comment by jaymcnallie** [comment]
:+1:

**Comment by jlcsmith** [comment]
Hero'd this PR.  Clean build was good.  Manually tested preferences including features and layers.  Everything looked good in both maps.

**Comment by jaymcnallie** [comment]
This was merged on the command line.  Closing this PR.

**Pull request closed without merging**